integration_test/tests/power_bi_tests.rs

```diff
@@ -244,4 +245,149 @@ mod integration {
+    ///     - SQLGetInfoW(SQL_DRIVER_NAME)
+    ///     - SQLExecDirectW({NullTerminatedQuery},SQL_NTS)
+    ///     - SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+    ///     - SQLGetInfoW(SQL_GETDATA_EXTENSIONS)
+    ///     - SQLNumResultCols()
+    ///     - For columns 1 to {numCols}
+    ///         - SQLColAttributeW(SQL_DESC_CONCISE_TYPE)
+    ///         - SQLColAttributeW(SQL_DESC_UNSIGNED)
+    ///         - SQLColAttributeW(SQL_COLUMN_NAME)
+    ///         - SQLColAttributeW(SQL_COLUMN_NULLABLE)
+    ///         - SQLColAttributeW(SQL_DESC_TYPE_NAME)
+    ///         - SQLColAttributeW(SQL_COLUMN_LENGTH)
+    ///         - SQLColAttributeW(SQL_COLUMN_SCALE)
+    ///     - For X rows or until SQLFetch returns SQL_NO_DATA
+    ///         - SQLFetch()
+    ///         - For columns 1 to {numCols}
+    ///             - SQLGetData({colIndex}, {defaultCtoSqlType})
+    ///     - SQLMoreResults()
+    ///     - SQLFreeHandle(SQL_HANDLE_STMT)
+    #[test]
+    fn test_data_retrieval() {
+        let env_handle: HEnv = setup();
+        let (conn_handle, _, _, _) = power_bi_connect(env_handle);
+        let mut stmt: Handle = null_mut();
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+
+            // SQL_DRIVER_ODBC_VER and SQL_DRIVER_NAME are not available through odbc_sys
+            /*
+            SQLGetInfoW(SQL_DRIVER_ODBC_VER)
+            SQLGetInfoW(SQL_DRIVER_NAME)
+            */
+
+            let mut query: Vec<u16> = "select * from example".encode_utf16().collect();
+            query.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), SQL_NTS as i32)
+            );
+
+            // SQLGetFunctions is not available through odbc_sys
+            /*
+            SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+            */
+
+            let str_len_ptr = &mut 0;
+            const BUFFER_LENGTH: SmallInt = 300;
+            let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetInfoW(
+                    conn_handle,
+                    InfoType::GetDataExtensions,
+                    output_buffer as Pointer,
+                    BUFFER_LENGTH,
+                    str_len_ptr
+                )
+            );
+
+            let column_count_ptr = &mut 0;
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLNumResultCols(stmt as HStmt, column_count_ptr)
+            );
+            let numeric_attribute_ptr = &mut 0;
+            const FIELD_TYPES: [Desc; 4] = [
+                Desc::ConciseType,
+                Desc::Unsigned,
+                // SQL_COLUMN_NAME,
```

Comment:
I want to flag the `SQLColAttribute` calls, specifically the `SQL_COLUMN_*` field types. From the [comments section of the spec] (https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolattribute-function?source=recommendations&view=sql-server-ver16), it seems these are ODBC 2.x values that ODBC 3.x is expected to handle/map appropriately. If I am reading / understanding them correctly, I believe they don't exist in the [odbc_sys Desc enum](https://github.com/pacman82/odbc-sys/blob/master/src/desc.rs#L199). I see for some `SQLGetInfo` calls we ignore if an enum value is unavailable (and I do that in a couple places too). I wanted to check if that is appropriate here as well for the column attribute calls, or if we should consider opening a ticket.

Reply:
Looking through my SQL trace log I see the ODBC 3.x field types being used: SQL_DESC_NAME, SQL_DESC_NULLABLE, SQL_DESC_SCALE, SQL_DESC_LENGTH.  I'm wondering if the flows were captured with an earlier version of our driver before we were reporting and ODBC version.  Are you still seeing these ODBC 2.x values @nbagnard ?  If not, then we should be able to use the SQL_DESC_* values as I see the ones we want here are available in odbc_sys.

Reply:
The flows were captured with Patrick's POC and I just checked it does not report the SQL_DRIVER_ODBC_VERSION, so it is possible that it defaults to 2.0.
I also see the ODBC 3.x identifiers SQL_DESC_SCALE, SQL_DESC_LENGTH, SQL_DESC_NAME being used now with a recent trace.
I think that you can safely switch to using only SQL_DESC_XXX identifiers here.

Reply:
I will update these to the 3.0 versions. Thanks for the feedback